### PR TITLE
Add reminders endpoints

### PR DESCRIPTION
**Pull request by amorriscode**
### tl;dr

- [x] Setups up `ConfigModule` to parse `.env` files
- [x] Adds CRUD routes for reminders
- [x] Connects reminders CRUD routes to Supabase
- [x] Converts project to use `fastify` (performance gains, why not? 😈)
- [x] Adds health check endpoint and removes unnecessary app controller/service

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -1,8 +1,22 @@
...
 import { AppModule } from './app.module';
 
 async function bootstrap() {
-  const app = await NestFactory.create(AppModule);
+  const app = await NestFactory.create<NestFastifyApplication>(
+    AppModule,
+    new FastifyAdapter(),
+  );
+
+  app.setGlobalPrefix('/v1');
```
Comment (amorriscode) [inline src/main.ts:15]: This is a good "best practice" so I just threw it in there. Basically, we'll namespace our APIs so we can do version control.

File: src/reminders/dto/index.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export * from './create-reminder.dto';
+export * from './update-reminder.dto';
```
Comment (amorriscode) [inline src/reminders/dto/index.ts:2]: Having an `index.ts` allows us to import all the DTOs for a single resource like this:

```typescript
import { CreateReminderDto, UpdateReminderDto } from './dto';
```

File: src/reminders/dto/update-reminder.dto.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { PartialType } from '@nestjs/mapped-types';
+import { CreateReminderDto } from './create-reminder.dto';
+
+export class UpdateReminderDto extends PartialType(CreateReminderDto) {
+  title: string;
+  description: string;
+  notificationDays: number[];
+  notificationTime: Date;
```
Comment (alicelovescake) [inline src/reminders/dto/update-reminder.dto.ts:8]: Reminder to change this time to string or timestamp format. So this is assuming they can only choose a single time for all days? i.e I want a gym reminder 8am Monday- friday. They can't have different times for diff days?

File: src/reminders/entities/reminder.entity.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export class Reminder {
+  id: number;
+  title: string;
+  description: string;
+  notificationDays: number[];
```
Comment (alicelovescake) [inline src/reminders/entities/reminder.entity.ts:5]: Should notification days be  numbers or a strings? Are you representing Monday as 1 and Tuesday as 2?

File: src/reminders/dto/update-reminder.dto.ts (hunk 3 above)
Comment (amorriscode) [reply src/reminders/dto/update-reminder.dto.ts:8]: Nah, let's keep it simple and just have one time per reminder.

File: src/reminders/entities/reminder.entity.ts (hunk 4 above)
Comment (amorriscode) [reply src/reminders/entities/reminder.entity.ts:5]: I figured we could go with numbers. 1 is Monday, as you said, through to 7 (Sunday). What do you think?

**Review by alicelovescake — APPROVED** [review]

**Pull request merged**